import_export/admin.py

```diff
@@ -600,22 +600,17 @@ def get_export_form(self):
             Use :meth:`~import_export.admin.ExportMixin.get_export_form_class` or set the new
             :attr:`~import_export.admin.ExportMixin.export_form_class` attribute.
         """
+        warnings.warn(
+            "ExportMixin.get_export_form() is deprecated and will be removed in "
+            "a future release. Please use get_export_form_class() instead.",
+            category=DeprecationWarning
+        )
         return self.export_form_class
 
     def get_export_form_class(self):
         """
         Get the form class used to read the export format.
         """
-        if not getattr(self.get_export_form, 'is_original', False):
-            warnings.warn(
-                "ExportMixin.get_export_form_class() is deprecated and will "
-                "be removed in a future release. Please use the new "
-                "`export_form_class` attribute to specify a custom form "
-                "class, or override the get_export_form_class() method if "
-                "your requirements are more complex.",
-                category=DeprecationWarning
-            )
-            return self.get_confirm_import_form()
```

Comment:
I think we need to restore this in order to keep things backwards compatibility... there are just a couple of tweaks needed (lazy copy and paste job, sorry!) 

 Without this, `get_export_form()` will never be called, as `export_action()` calls this method only